test/powershell/engine/Api/TypeInference.Tests.ps1

```diff
@@ -1469,6 +1469,26 @@ Describe "Type inference Tests" -tags "CI" {
         $res.Count | Should -Be 0
     }
 
+    It 'Infers right side of assignment expression' {
+        $res = [AstTypeInference]::InferTypeOf( { $Test1 = "Hello" }.Ast.Find({param($ast) $ast -is [Language.AssignmentStatementAst]}, $true))
+        $res.Count | Should -Be 1
+        $res.Name | Should -Be "System.String"
+    }
+
+    It 'Infers left side of assignment expression when it is a ConvertExpression' {
+        $res = [AstTypeInference]::InferTypeOf( { [string]$Test1 = 42 }.Ast.Find({param($ast) $ast -is [Language.AssignmentStatementAst]}, $true))
+        $res.Count | Should -Be 1
+        $res.Name | Should -Be "System.String"
+    }
+
+    It 'Infers left side of assignment expression when there is a ConvertExpression among other attributes' {
+        $res = [AstTypeInference]::InferTypeOf( {
+            [ValidateLength()] [string] [ValidatePattern()]$Test1 = 42
+        }.Ast.Find({param($ast) $ast -is [Language.AssignmentStatementAst]}, $true))
+        $res.Count | Should -Be 1
+        $res.Name | Should -Be "System.String"
+    }
+
```

Comment:
I don't see tests cases from the PR description:
```
This will fix the type inference for multiple variable assignments like: $Test1 = [string]$Test2 = $Test3 = $Test4 = 10 and also variables assigned with parentheses: $Test1 = ($Test2 = ls)
```

Reply:
That's because the examples from the description are practical examples of where the effect of this change can be felt. However, since the API lets us specify the exact Ast we want to infer, there is no need for the practical examples in the tests as we can make them more specific.